how-to-guides/e2e-tracking/scripts/e2e_tracking.py

```diff
@@ -0,0 +1,307 @@
+# How to track models end-to-end on Neptune? #
+# ========================================== #
+#
+# This script shows how you can use Neptune to track a model across all stages of it's lifecycle by:
+# * Logging model and run metadata to a central project
+# * Comparing runs to select the best performing model
+# * Monitoring a model once in production
+#
+# This script can be used as a template to design an automated end-to-end pipeline that covers the
+# entire lifecycle of a model without needing any manual intervention.
+#
+# This example uses Optuna hyperparameter-optimization to simulate training and evaluating multiple
+# scikit-learn models, and Evidently to monitor models in production.
+# However, given Neptune's flexibility and multiple integrations, you can use any library and
+# framework of your choice.
+#
+# List of all Neptune integrations: https://docs.neptune.ai/integrations/
+
+
+# ===== Before you start ===== #
+# This script example lets you try out Neptune anonymously, with zero setup.
+# If you want to see the example logged to your own workspace instead:
+# 1. Create a Neptune account --> https://neptune.ai/register
+# 2. Create a Neptune project that you will use for tracking metadata.
+#    Instructions --> https://docs.neptune.ai/setup/creating_project
+
+## Import dependencies
+
+import os
+import pickle as pkl
+import time
+
+import matplotlib
+import neptune
+import numpy as np
+import optuna
+from evidently.metric_preset import RegressionPreset
+from evidently.metrics import *
+from evidently.report import Report
+from neptune.exceptions import NeptuneModelKeyAlreadyExistsError
+from neptune.integrations.optuna import NeptuneCallback
+from neptune.integrations.sklearn import create_regressor_summary, get_pickled_model
+from neptune.utils import stringify_unsupported
+from sklearn.datasets import fetch_california_housing
+from sklearn.ensemble import RandomForestRegressor
+from sklearn.model_selection import train_test_split
+
+# To prevent `RuntimeError: main thread is not in main loop` error
+matplotlib.use("Agg")
+
+# ===== Track model training ===== #
+# 1. Use Optuna to train multiple scikit-learn regression models
+# 2. Leverage Neptune's scikit-learn and Optuna integrations to automatically log metadata and metrics
+#    to Neptune for easy run comparison, while also using Neptune's model registry to track models.
+
+## Prepare the dataset ##
+
+data, target = fetch_california_housing(return_X_y=True)
+X_train, X_test, y_train, y_test = train_test_split(data, target, test_size=0.25)
+
+### Initialize a Neptune project
+# This step assumes that your Neptune API token is set as an environment variable
+# **Note:** The "common/e2e-tracking" project used here is read-only.
+# To log to your own project, just update the `NEPTUNE_PROJECT` environment variable.
+
+# Comment out the below line if your `NEPTUNE_PROJECT` env variable is already set
+os.environ["NEPTUNE_PROJECT"] = "common/e2e-tracking"
```

Comment:
IMO it would be better to have this commented and require uncommenting in order to set the environment variable through the script.

Reply:
This is needed for tests and e2e hands-free runs.
I can set the env variables through `run_examples.sh`, but then the script would fail if someone downloaded it and tried to run it locally

Reply:
I just feel hesitant about providing an example script that edits system variables unless you modify the code. It feels too pushy, as we can't easily provide a way for them to restore the variable to the user's own credentials if they change them by mistake. (Of course it's not wise to grab a script and execute it without knowing what it does, but we don't want to counter malpractice with malpractice.)

At the very least, we should do the env var stuff at the top of the script so it's as transparent and clear as possible.

@parthpankajtiwary Could you give your take?

Reply:
(If Parth is too busy to comment by EoD, let's proceed with the other approach you mentioned @SiddhantSadangi)

Reply:
Environment variables updated in scripts using `%env` or `os.environ` persist only as long as the script runs.
Once the script terminates, the values revert to what they were before

Reply:
Moving it to the top would be better than removing it completely though

Reply:
Ah, in that case it's not so bad. But, I'm still in favor of frontloading the parts where it would be best for users to fill in their own data.

Reply:
Moved it up as much as I think I can without sacrificing code quality.. lemme know what you think

Reply:
> are the users able to log to this project (common/e2e-tracking) without any changes to the script?

Yup. Using the Anonymous token